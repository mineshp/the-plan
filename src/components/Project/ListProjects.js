import React from 'react';
import PropTypes from 'prop-types';
import { Card, Container } from 'semantic-ui-react';
import DisplayMessage from '../Shared/DisplayMessage';

const ListProjects = ({
    errors,
    cards
}) => (
    <Container>
        {
            (errors && errors.error) &&
                <DisplayMessage status={errors} />
        }
        <div className="List main">
            <Card.Group itemsPerRow={3}>
                {cards}
            </Card.Group>
        </div>
    </Container>
);

ListProjects.propTypes = {
    errors: PropTypes.shape({}),
    cards: PropTypes.arrayOf(
        PropTypes.shape({})
    )
};

ListProjects.defaultProps = {
    errors: null,
    cards: []
};

export default ListProjects;
