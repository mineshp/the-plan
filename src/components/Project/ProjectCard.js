import React from 'react';
import PropTypes from 'prop-types';
import { Button, Card } from 'semantic-ui-react';
import { formatDate } from '../../helpers/validators/common';

const ProjectCard = ({ data, onDeleteHandler }) => (
    <Card color={data.colour}>
        <Card.Content>
            <Card.Header className="firstletter-capitalise">
                {data.projectName}
                <Button
                    floated="right"
                    icon="list layout"
                    color="blue"
                    href={`/project/${data.projectName}/lists`} // eslint-disable-line no-underscore-dangle
                    title={`View all lists for project ${data.projectName}`}
                />
            </Card.Header>
        </Card.Content>
        <Card.Content>
            <Card.Meta>
                Created by: {data.owner}
            </Card.Meta>
            <Card.Meta>
                {formatDate(data.createdDate)}
            </Card.Meta>
            <Card.Description>
                {data.projectDescription}
            </Card.Description>
        </Card.Content>
        <Card.Content extra>
            <div className="buttons">
                <Button
                    as="a"
                    content="Update"
                    icon="edit"
                    labelPosition="left"
                    color="green"
                    href={`/project/update/${data._id}`} // eslint-disable-line no-underscore-dangle
                />
                <Button
                    content="Delete"
                    icon="trash"
                    labelPosition="left"
                    color="pink"
                    value={data._id} // eslint-disable-line no-underscore-dangle
                    onClick={onDeleteHandler}
                />
            </div>
        </Card.Content>
    </Card>
);

ProjectCard.propTypes = {
    data: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        projectName: PropTypes.string.isRequired,
        colour: PropTypes.string
    }).isRequired,
    onDeleteHandler: PropTypes.func.isRequired

};

export default ProjectCard;
