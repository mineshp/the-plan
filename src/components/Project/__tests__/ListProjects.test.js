import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ProjectCard from '../ProjectCard';
import ListProjects from '../ListProjects';
import { mockProjectData } from '../../../helpers/test/testData/projectData';

const mockResult = {};
const mockErrorResult = {
    error: {
        message: 'oh no something bad happened',
        isError: true
    }
};
const mockProject = mockProjectData();

const mockHandleDelete = jest.fn();
// eslint-disable-next-line no-underscore-dangle
const mockProjectId = mockProject._id;
const mockCards = [
    <MemoryRouter key={mockProjectId}>
        <ProjectCard data={mockProject} key={mockProjectId} onDeleteHandler={mockHandleDelete} />
    </MemoryRouter>
];

describe('List Projects', () => {
    it('renders projects correctly', () => {
        const tree = renderer.create(<ListProjects
            errors={mockResult}
            cards={mockCards}
            onDeleteHandler={mockHandleDelete}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders successfully after successful creation of project', () => {
        const tree = renderer.create(<ListProjects
            errors={mockResult}
            cards={mockCards}
            onDeleteHandler={mockHandleDelete}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders successfully after error creating new project', () => {
        const tree = renderer.create(<ListProjects
            errors={mockErrorResult}
            cards={mockCards}
            onDeleteHandler={mockHandleDelete}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
