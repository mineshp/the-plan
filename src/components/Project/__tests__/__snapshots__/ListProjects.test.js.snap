// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Projects renders projects correctly 1`] = `
<div
  className="ui container"
>
  <div
    className="List main"
  >
    <div
      className="ui three cards"
    >
      <div
        className="ui card"
        href={undefined}
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            test
          </div>
        </div>
        <div
          className="content"
        >
          <div
            className="meta"
          />
          <div
            className="description"
          >
            Add a description here
          </div>
        </div>
        <div
          className="extra content"
        >
          <div
            className="buttons"
          >
            <a
              className="ui teal icon left labeled button"
              disabled={undefined}
              href="/project/update/1"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              <i
                aria-hidden="true"
                className="edit icon"
              />
              Update
            </a>
            <button
              className="ui blue icon left labeled button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              value="1"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`List Projects renders successfully after error creating new project 1`] = `
<div
  className="ui container"
>
  <div
    className="ui red message"
  >
    <p>
      oh no something bad happened
    </p>
  </div>
  <div
    className="List main"
  >
    <div
      className="ui three cards"
    >
      <div
        className="ui card"
        href={undefined}
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            test
          </div>
        </div>
        <div
          className="content"
        >
          <div
            className="meta"
          />
          <div
            className="description"
          >
            Add a description here
          </div>
        </div>
        <div
          className="extra content"
        >
          <div
            className="buttons"
          >
            <a
              className="ui teal icon left labeled button"
              disabled={undefined}
              href="/project/update/1"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              <i
                aria-hidden="true"
                className="edit icon"
              />
              Update
            </a>
            <button
              className="ui blue icon left labeled button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              value="1"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`List Projects renders successfully after successful creation of project 1`] = `
<div
  className="ui container"
>
  <div
    className="List main"
  >
    <div
      className="ui three cards"
    >
      <div
        className="ui card"
        href={undefined}
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            test
          </div>
        </div>
        <div
          className="content"
        >
          <div
            className="meta"
          />
          <div
            className="description"
          >
            Add a description here
          </div>
        </div>
        <div
          className="extra content"
        >
          <div
            className="buttons"
          >
            <a
              className="ui teal icon left labeled button"
              disabled={undefined}
              href="/project/update/1"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              <i
                aria-hidden="true"
                className="edit icon"
              />
              Update
            </a>
            <button
              className="ui blue icon left labeled button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              value="1"
            >
              <i
                aria-hidden="true"
                className="trash icon"
              />
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
